docs/python-examples/fixtures/conftest.py

```diff
@@ -0,0 +1,109 @@
+
+    def fin():
+        if not sc.is_deleted:
+            sc.delete()
+
+    request.addfinalizer(fin)
+    return sc_name
+
+
+@pytest.fixture(scope='class')
+def cls_pvc(request, storage_class):
+    """
+    PVC fixture
+
+    Returns:
+        PVC: object of pvc class
+    """
+    pvc_name = ocs.get_random_name('pvc')
+    pvc = ocs.PVC(pvc_name, storage_class)
+    logger.info(f"Creating pvc: {pvc.name}")
+
+    def fin():
+        if not pvc.is_deleted:
+            pvc.delete()
+
+    request.addfinalizer(fin)
+    return pvc
+
+
+@pytest.fixture(scope='class')
+def pvcs_factory(request, storage_class):
+    """
+    Fixture factory for creating pvcs.
+    This fixture is returning function with with which you can create objects
+    of PVC as a part of test and cares about teardown of created PVCs.
+
+    Usage of this fixter is for usecases when:
+    * you need to create PVC resources as part of test and the teardown is
+      done automatically by its teardown
+    * you need to create PVC resources as part of test and you can also delete
+      as part of test, thanks to is_deleted flag the teardown is skipped for
+      already deleted PVCs
+    * if you need to prepare those resources as part of setup fixture, please
+      create wrapper for this factory fixture like will be showed in another
+      example.
+
+    Returns:
+        function wrapper wrapper_crate_pvcs for ocs.create_pvcs function.
+    """
+    # this is list of pvcs where we will append all created pvcs by helper
+    # function for creating PVC and will be used in finalizer as well.
+    pvcs = []
+    logger.info("Setup of pvcs")
+
+    # the pvcs=pvcs reference needs to be here otherwise it's not visible
+    # from scope of class/method from which we will call the wrapper function
+    # returned by this fixture factory.
+    # There is possibility to define helper function inside this factory like
+    # in example:
+    # https://docs.pytest.org/en/latest/fixture.html#factories-as-fixtures
+    # But then we will use capability of rusability of code. So this solution
+    # should work for us. Another solution is using decorator but this is more
+    # comples, otherwise I didn't come with other solution how to reuse easily
+    # code in our fixture factories.
+    def wrapper_crate_pvcs(count, storage_class, pvcs=pvcs):
+        """
+        Function wrapper for ocs.create_pvcs. This wrapper append created PVCs
+        into pvcs list referenced from pvcs_factory which allows to do proper
+        teardown of created objects.
+
+        Args:
+            count (int): desired count of pvcs to be created
+            storage_class (StorageClass): reference to storage class object
+            pvcs (list): reference to pvcs list where to append all created
+                pvcs by helper function ocs.create_pvcs
+        """
+        ret_value = ocs.create_pvcs(count, storage_class)
+        pvcs += ret_value
```

Comment:
```suggestion
        pvcs.append(ret_value)
```

Reply:
Ret value is a list of pvcs, cannot do append otherwise pvcs will look like [[pvc1, pVc2]] and after second call [[pvc1, pvc2],[pvc3, pvc4, pvc5]] for example.

Reply:
Ah, good call